### PR TITLE
ci: pin lib-metadata for appsec to avoid otel dependency error

### PR DESCRIPTION
**Pull request by ZStriker19**
## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [ ] Title is accurate.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Testing strategy adequately addresses listed risk(s).
- [ ] Change is maintainable (easy to change, telemetry, documentation).
- [ ] Release note makes sense to a user of the library.
- [ ] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: riotfile.py [hunk 1]
```diff
@@ -319,13 +320,16 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
         Venv(
             name="telemetry",
             command="pytest {cmdargs} tests/telemetry/",
-            pys=select_pys(),
+            # otel doesn't support python 2.7 and has changes in 3.5
```
Comment (mabdinur) [inline riotfile.py:323]: otel does not support `python<3.6` but instrumentation telemetry should support and test all python versions. 

Also why do we need to import the opentelemetry package in instrumentation telemetry tests? Instrumentation telemetry should be independent.

Comment (ZStriker19) [reply riotfile.py:323]: We're importing the otel package to generate otel traces to test span creation metrics.

File: riotfile.py [hunk 2]
```diff
@@ -319,13 +320,16 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+            pys=select_pys(min_version="3.7"),
             pkgs={
                 # httpretty v1.0 drops python 2.7 support
                 "requests": latest,
                 "gunicorn": latest,
                 "flask": "<=2.2.3",
                 "httpretty": "==0.9.7",
+                "opentelemetry-api": latest,
+                "opentracing": latest,
```
[the remark this answers is not in the document]
Comment (ZStriker19) [reply riotfile.py:332]: I was getting dependency issues before this, but can try removing.

File: riotfile.py (hunk 1 above)
Comment (ZStriker19) [reply riotfile.py:323]: Ah sorry this got added from another PR

File: riotfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZStriker19) [reply riotfile.py:332]: wil remove

**Pull request closed without merging**